src/core/infer.ml

```diff
@@ -41,8 +41,72 @@ let unif : problem -> octxt -> term -> term -> unit =
 
 (** {1 Handling coercions} *)
 
-(* Simply unify types, no coercions yet. *)
-let coerce pb c t a b = unif pb c a b; (t, false)
+(** [simplify c t] tries to reduce coercions that are in term [t]. The
```

Comment:
simplify -> reduce_coercions (to avoid possible confusion with other reduction funtions)